### PR TITLE
ATR-759: Add release notes for 3.1.2, update 1007 diagnostic description

### PR DESCRIPTION
**Pull request by EAndrosova**

### CODE REVIEW

File: docs/ReleaseNotes.md [hunk 1]
```diff
@@ -1,6 +1,32 @@
...
+ 
+### Fixed Bugs
+In this version of Acuminator, the following bugs have been fixed:
+ - The [PX1023](diagnostics/PX1023.md) diagnostic showed false alert for the `INUnit` attribute.
+ - In Visual Studio 2022, Acuminator analysis did not work in the out-of-process mode (the **Run code analysis in separate process** check box in Visual Studio options)
+ - It was possible to suppress the [PX1007](diagnostics/PX1007.md) diagnostic for a class with the `PXCacheExtension` attribute by marking the class with the `PXHidden` attribute
+ - The [PX1016](diagnostics/PX1016.md) diagnostic showed a false alert for generic non-abstract graph extensions.
+
+## Gratitudes 
+The Acuminator team would like to thank our open source contributor @taskinkemal. We are grateful for your input and hope you stay with us.
```
Comment (SENya1990) [inline docs/ReleaseNotes.md:27]: Imho, we should mention a direct name together with the github account, it's Kemal Taskin.

File: docs/ReleaseNotes.md [hunk 2]
```diff
@@ -1,6 +1,32 @@
...
+In Acuminator 3.1.2, the following enhancements have been implemented:
+ - The [PX1096](diagnostics/PX1096.md) diagnostic has been added to check signature of methods with the `PXOverride` attribute.
+ - The [PX1007](diagnostics/PX1007.md) diagnostic no longer checks the following DAC fields for a code annotation:
+   - `Selected`
+   - `Attributes`
+   - `GroupMask`
+   - `CompanyID`
+   - `CompanyMask`
+   - `DeletedDatabaseRecord`
+ - The [PX1007](diagnostics/PX1007.md) diagnostic now checks that mapped DAC fields of projection DACs have the `inheritoc` XML tag
```
Comment (SENya1990) [inline docs/ReleaseNotes.md:17]: That's not complete description + there is a mistype in the tag name. 
PX1007 now checks that mapped DAC field **properties** have `inheritdoc` tag with `cref` attribute pointing to the mapped DAC field property of the original DAC. An example of correctly declared projection DAC property will be really helpful here and a link to the inheritdoc tag documentation:
https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/xmldoc/recommended-tags#inheritdoc

File: docs/ReleaseNotes.md [hunk 3]
```diff
@@ -1,6 +1,32 @@
...
+
```
Comment (SENya1990) [inline docs/ReleaseNotes.md:18]: The PX1007 changes description should also include that PX1007 supports  `inheritdoc` tag in general, not just for projection DAC mapped properties. Basically, you can put any form of inheritdoc tag on normal DACs, DAC extensions and their properties but you have to follow particular rule for projection DAC mapped DAC properties.

Comment (SENya1990) [inline docs/ReleaseNotes.md:18]: There is also no description for a new code fix which will generate correct inheritdoc tag for a mapped DAC property of a projection DAC. It's worth mentioning that VS allows us to apply single code fix to multiple alerts with its "Fix All" functionality for code fixes. It means that this code fix can be used to very quickly automatically generate correct documentation for most of the projection DAC properties. Before we had to do it manually which is a rather tedious task.

Comment (SENya1990) [inline docs/ReleaseNotes.md:18]: Another change that is worth mentioning is that now all PX1007 code fixes generate documentation correctly even for a badly formatted code, and will keep existing documentation tags (except the tags deleted by the inheritdoc tag generation code fix I described in the previous comment)

Also the diagnostic and code fixes now support doc tags declared as empty XML elements like `<summary/>`

You can use the PR overview for the list of al PX1007 changes https://github.com/Acumatica/Acuminator/pull/497

File: docs/ReleaseNotes.md [hunk 4]
```diff
@@ -1,6 +1,32 @@
 # Acuminator Release Notes
 This document provides information about fixes, enhancements, and key features that are available in Acuminator.
 
+## Acuminator 3.1.2
+Acuminator 3.1.2 includes the bug fixes and enhancements described in this section, as well as the features that have been implemented in previous versions.
+
+### Enhancements
+In Acuminator 3.1.2, the following enhancements have been implemented:
+ - The [PX1096](diagnostics/PX1096.md) diagnostic has been added to check signature of methods with the `PXOverride` attribute.
```
Comment (SENya1990) [inline docs/ReleaseNotes.md:9]: I would rephrase this a bit, it's not just a some kind of generic unspecified check of PXOverrides. It's a check that in the base graph and base graph extensions there exists a base method that is compatible with the PXOverride method. This means that the base method has the same name, its accessibility public, protected or protected internal, and it has a compatible parameters.  More about this here:
- https://help.acumatica.com/(W(147))/Help?ScreenId=ShowWiki&pageid=6fa2a444-17b4-42f9-9e6a-64e85167626a
- https://help.acumatica.com/(W(146))/Help?ScreenId=ShowWiki&pageid=635c830e-4617-4d5c-9fa5-035952311aa9

File: docs/ReleaseNotes.md [hunk 5]
```diff
@@ -1,6 +1,32 @@
 # Acuminator Release Notes
 This document provides information about fixes, enhancements, and key features that are available in Acuminator.
 
+## Acuminator 3.1.2
```
Comment (SENya1990) [inline docs/ReleaseNotes.md:4]: It would be good to start adding dates to releases

File: docs/ReleaseNotes.md [hunk 6]
```diff
@@ -1,6 +1,32 @@
...
+   - `GroupMask`
+   - `CompanyID`
+   - `CompanyMask`
+   - `DeletedDatabaseRecord`
+ - The [PX1007](diagnostics/PX1007.md) diagnostic now checks that mapped DAC fields of projection DACs have the `inheritoc` XML tag
+ 
+### Fixed Bugs
+In this version of Acuminator, the following bugs have been fixed:
+ - The [PX1023](diagnostics/PX1023.md) diagnostic showed false alert for the `INUnit` attribute.
+ - In Visual Studio 2022, Acuminator analysis did not work in the out-of-process mode (the **Run code analysis in separate process** check box in Visual Studio options)
```
Comment (SENya1990) [inline docs/ReleaseNotes.md:22]: Maybe we should emphasize this bug a little? Maybe place it first? It's actually a blocker that was a reason for an early release.

File: docs/ReleaseNotes.md [hunk 7]
```diff
@@ -1,6 +1,32 @@
...
+ - It was possible to suppress the [PX1007](diagnostics/PX1007.md) diagnostic for a class with the `PXCacheExtension` attribute by marking the class with the `PXHidden` attribute
```
Comment (SENya1990) [inline docs/ReleaseNotes.md:23]: This is not correct. `PXCacheExtension` is not an attribute, its a base type for DAC extensions. Hence sometimes DAC extensions are also called cache extensions. The more correct version would be something like:
```
 It was possible to suppress the [PX1007](diagnostics/PX1007.md) diagnostic for DAC extensions by marking them with the `PXHidden` attribute
```

File: docs/diagnostics/PX1007.md [hunk 8]
```diff
@@ -8,7 +8,10 @@ This document describes the PX1007 diagnostic.
 | PX1007 | A DAC, a DAC extension, and public DAC properties should have a description in the `summary` XML tag. | Warning | Available | 
 
 ## Diagnostic Description
-Any DAC or DAC extension and their public property fields should have a description in the `summary` XML tag. The descriptions of the DACs, DAC extensions, and their members can be displayed in the DAC Schema Browser and used for building the API and DAC references for the Acumatica ERP libraries.
+Any DAC or DAC extension and their public property fields should have one of the following:
```
Comment (SENya1990) [inline docs/diagnostics/PX1007.md:11]: There should be many more details, the diagnostic became rather complex. You can find them here:
https://github.com/Acumatica/Acuminator/pull/497#issue-1722660403

The documentation should mention that PX1007 now supports one or multiple inheritdoc tags in any format on DACs, DAC extensions and DAC field properties of non projection DACs (and even unmapped properties of projection DACs). 
But it has strict rules regarding the mapped properties of projection DACs.

File: docs/diagnostics/PX1007.md [hunk 9]
```diff
@@ -8,7 +8,10 @@ This document describes the PX1007 diagnostic.
...
+* A description in the `summary` XML tag
+* The 'inheritoc' XML tag on mapped DAC fields of projection DACs
```
Comment (SENya1990) [inline docs/diagnostics/PX1007.md:13]: exclude tag is not in the list.

File: docs/diagnostics/PX1007.md [hunk 10]
```diff
@@ -29,7 +32,22 @@ The diagnostic is not displayed for the following items:
  - Members of public entities (except nested public types and DAC property fields).
  - All members of a public entity if the entity has the `exclude` XML tag or is marked with the `PXHidden`, `PXInternalUseOnly`, or `Obsolete` attribute.
  - Class fields of DACs. 
- - System fields of DACs, which are the following: `CreatedByID`, `CreatedByScreenID`, `CreatedDateTime`, `LastModifiedByID`, `LastModifiedByScreenID`, `LastModifiedDateTime`, `TStamp`, and `NoteID`.
+ - Well-known DAC fields, which are the following:
+   - `Selected`
+   - `Attributes`
```
Comment (SENya1990) [inline docs/diagnostics/PX1007.md:37]: Attributes and GroupMask are system fields with the special role in the platform

File: docs/diagnostics/PX1007.md [hunk 11]
```diff
@@ -29,7 +32,22 @@ The diagnostic is not displayed for the following items:
...
+   - `CreatedByID`
+   - `CreatedByScreenID`
+   - `CreatedDateTime`
+   - `LastModifiedByID`
+   - `LastModifiedByScreenID`
+   - `LastModifiedDateTime`
+   - `TStamp`
+   - `NoteID`.
 
 If you suppress this diagnostic with a comment, the default comment message is the following: _Acuminator disable once PX1007 NoXmlCommentForPublicEntityOrDacProperty to be documented later_.
```
Comment (SENya1990) [inline docs/diagnostics/PX1007.md:52]: The default justification was deleted. Developers shouldn't suppress the diagnostic to exclude API from the documentation. They should use exclude tag for this.

File: docs/ReleaseNotes.md [hunk 12]
```diff
@@ -1,6 +1,32 @@
 # Acuminator Release Notes
 This document provides information about fixes, enhancements, and key features that are available in Acuminator.
 
+## Acuminator 3.1.2
+Acuminator 3.1.2 includes the bug fixes and enhancements described in this section, as well as the features that have been implemented in previous versions.
+
+### Enhancements
+In Acuminator 3.1.2, the following enhancements have been implemented:
+ - The [PX1096](diagnostics/PX1096.md) diagnostic has been added to check signature of methods with the `PXOverride` attribute.
+ - The [PX1007](diagnostics/PX1007.md) diagnostic no longer checks the following DAC fields for a code annotation:
```
Comment (SENya1990) [inline docs/ReleaseNotes.md:10]: The default justification for PX1007 was deleted, this should be mentioned.

File: docs/diagnostics/PX1007.md [hunk 13]
```diff
@@ -8,7 +8,10 @@ This document describes the PX1007 diagnostic.
 | PX1007 | A DAC, a DAC extension, and public DAC properties should have a description in the `summary` XML tag. | Warning | Available | 
 
 ## Diagnostic Description
-Any DAC or DAC extension and their public property fields should have a description in the `summary` XML tag. The descriptions of the DACs, DAC extensions, and their members can be displayed in the DAC Schema Browser and used for building the API and DAC references for the Acumatica ERP libraries.
+Any DAC or DAC extension and their public property fields should have one of the following:
+* A description in the `summary` XML tag
+* The 'inheritoc' XML tag on mapped DAC fields of projection DACs
+The descriptions of the DACs, DAC extensions, and their members can be displayed in the DAC Schema Browser and used for building the API and DAC references for the Acumatica ERP libraries.
 
 The code fix does one of the following:
```
Comment (SENya1990) [inline docs/diagnostics/PX1007.md:16]: No description for a new code fix that adds inheritdoc tag was added

File: docs/diagnostics/PX1007.md [hunk 14]
```diff
@@ -8,7 +8,10 @@ This document describes the PX1007 diagnostic.
 | PX1007 | A DAC, a DAC extension, and public DAC properties should have a description in the `summary` XML tag. | Warning | Available |
```
Comment (SENya1990) [inline docs/diagnostics/PX1007.md:8]: The overview should be fixed to mention inheritdoc tag. 
The same applies to the Summary.md file

**Review by SENya1990 — CHANGES_REQUESTED** [review]
The release notes and diagnostic documentation need more work

File: docs/ReleaseNotes.md (hunk 2 above)
Comment (EAndrosova) [reply docs/ReleaseNotes.md:17]: Is it okay if I give the link to inheritdoc documnetation and details about the original mapped property in the article about PX1007? This is too specific info for RNs

Comment (SENya1990) [reply docs/ReleaseNotes.md:17]: Yes, it's okay.

File: docs/ReleaseNotes.md (hunk 3 above)
Comment (EAndrosova) [reply docs/ReleaseNotes.md:22]: Could you plesae clarify what is a badly formatted code?

Comment (SENya1990) [reply docs/ReleaseNotes.md:22]: For example, the code without the empty lines between elements:
```csharp
#region OrderType
public abstract class orderType : IBqlField { }
[PXDBString(IsKey = true, InputMask = "")]
[PXDefault]
[PXUIField(DisplayName = "Order Type")]
public string OrderType { get; set; }
#endregion
```

File: docs/ReleaseNotes.md [hunk 15]
```diff
@@ -1,30 +1,34 @@
...
 This document provides information about fixes, enhancements, and key features that are available in Acuminator.
 
-## Acuminator 3.1.2
+## Acuminator 3.1.2 - 2023-06-06
 Acuminator 3.1.2 includes the bug fixes and enhancements described in this section, as well as the features that have been implemented in previous versions.
 
 ### Enhancements
 In Acuminator 3.1.2, the following enhancements have been implemented:
- - The [PX1096](diagnostics/PX1096.md) diagnostic has been added to check signature of methods with the `PXOverride` attribute.
+ - The [PX1096](diagnostics/PX1096.md) diagnostic has been added to check signature of methods with the `PXOverride` attribute: The diagnostic checks that the base graph or base graph extensions contains a base method that is compatible with the method with the `PXOverride` attribute.
```
Comment (SENya1990) [inline docs/ReleaseNotes.md:9]: Maybe rephrase it like this?

```suggestion
 - The [PX1096](diagnostics/PX1096.md) diagnostic has been added to check signature of methods with the `PXOverride` attribute: The diagnostic checks whether the base graph or base graph extensions contain a base method  compatible with the method with the `PXOverride` attribute. 
```

File: docs/ReleaseNotes.md [hunk 16]
```diff
@@ -1,30 +1,34 @@
...
  - The [PX1007](diagnostics/PX1007.md) diagnostic no longer checks the following DAC fields for a code annotation:
    - `Selected`
    - `Attributes`
    - `GroupMask`
    - `CompanyID`
    - `CompanyMask`
    - `DeletedDatabaseRecord`
- - The [PX1007](diagnostics/PX1007.md) diagnostic now checks that mapped DAC fields of projection DACs have the `inheritoc` XML tag
+ - The [PX1007](diagnostics/PX1007.md) diagnostic has been updated in the following ways:
+   - The diagnostic now supports the `inheritdoc` tag for DACs, DAC extensions, and their properties. The diagnostic also checks that mapped DAC field properties have the `inheritdoc` tag with the `cref` attribute pointing to the mapped DAC field property of the original DAC.
```
Comment (SENya1990) [inline docs/ReleaseNotes.md:18]:
```suggestion
   - The diagnostic now supports the `inheritdoc` tag for DACs, DAC extensions, and their properties. The diagnostic also checks that DAC field properties of projection DACs have the `inheritdoc` tag with the `cref` attribute pointing to the mapped DAC field property of the original DAC. 
```

File: docs/ReleaseNotes.md [hunk 17]
```diff
@@ -1,30 +1,34 @@
...
    - `GroupMask`
    - `CompanyID`
    - `CompanyMask`
    - `DeletedDatabaseRecord`
- - The [PX1007](diagnostics/PX1007.md) diagnostic now checks that mapped DAC fields of projection DACs have the `inheritoc` XML tag
+ - The [PX1007](diagnostics/PX1007.md) diagnostic has been updated in the following ways:
+   - The diagnostic now supports the `inheritdoc` tag for DACs, DAC extensions, and their properties. The diagnostic also checks that mapped DAC field properties have the `inheritdoc` tag with the `cref` attribute pointing to the mapped DAC field property of the original DAC. 
+   - A new code fix can generate the `inheritdoc` tag for a mapped DAC property of a projection DAC.
+   - Code fixes of the diagnostic can now generate documentation correctly even for a badly formatted code and remove incorrect documentation tags. The code fixes also support doc tags declared as empty XML elements such as `<summary/>`.
+   - The default justification "to be documented later" has been removed from the diagnostic. To remove the DAC or its property from the documentation, use the `exclude` tag.
```
Comment (SENya1990) [inline docs/ReleaseNotes.md:21]:
```suggestion
   - The default justification "to be documented later" has been removed from the diagnostic. To remove a DAC, DAC extension or DAC field property from the documentation, use the `exclude` tag.
```

File: docs/ReleaseNotes.md [hunk 18]
```diff
@@ -1,30 +1,34 @@
...
    - `Selected`
    - `Attributes`
    - `GroupMask`
    - `CompanyID`
    - `CompanyMask`
    - `DeletedDatabaseRecord`
- - The [PX1007](diagnostics/PX1007.md) diagnostic now checks that mapped DAC fields of projection DACs have the `inheritoc` XML tag
+ - The [PX1007](diagnostics/PX1007.md) diagnostic has been updated in the following ways:
+   - The diagnostic now supports the `inheritdoc` tag for DACs, DAC extensions, and their properties. The diagnostic also checks that mapped DAC field properties have the `inheritdoc` tag with the `cref` attribute pointing to the mapped DAC field property of the original DAC. 
+   - A new code fix can generate the `inheritdoc` tag for a mapped DAC property of a projection DAC.
```
Comment (SENya1990) [inline docs/ReleaseNotes.md:19]: It's worth mentioning that VS allows us to apply code fix quickly to multiple alerts with its "Fix All" functionality for code fix.

File: docs/ReleaseNotes.md (hunk 3 above)
Comment (SENya1990) [reply docs/ReleaseNotes.md:22]: this remark has not been covered yet

File: docs/diagnostics/PX1007.md [hunk 19]
```diff
@@ -5,18 +5,26 @@ This document describes the PX1007 diagnostic.
 
 | Code   | Short Description                                   | Type                             | Code Fix  | 
 | ------ | ----------------------------------------------------| -------------------------------- | --------- | 
-| PX1007 | A DAC, a DAC extension, and public DAC properties should have a description in the `summary` XML tag. | Warning | Available | 
+| PX1007 | The DAC, DAC extension, or DAC property should have a description in the "summary" XML tag or "inheritdoc" XML tag | Warning | Available |
```
Comment (SENya1990) [inline docs/diagnostics/PX1007.md:8]: We probably should change the type to `Error` because now PX1007 will report error for mapped DAC field properties of projection DACs with incorrect annotation:
```suggestion
| PX1007 | The DAC, DAC extension, or DAC property should have a description in the "summary" XML tag or "inheritdoc" XML tag | Error | Available | 
```

File: docs/Summary.md [hunk 20]
```diff
@@ -18,7 +18,7 @@ Acuminator does not perform static analysis of projects whose names contain `Tes
 | [PX1004](diagnostics/PX1004.md) | The order of view declarations will cause the creation of two cache instances. | Message | Unavailable |
 | [PX1005](diagnostics/PX1005.md) | There is probably a typo in the view delegate name. | Warning (ISV Level 3: Informational) | Available |
 | [PX1006](diagnostics/PX1006.md) | The order of view declarations will cause the creation of one cache instance for multiple DACs | Message | Unavailable |
-| [PX1007](diagnostics/PX1007.md) | A DAC, a DAC extension, and public DAC properties should have a description in the `summary` XML tag. | Warning | Available |
+| [PX1007](diagnostics/PX1007.md) | The DAC, DAC extension, or DAC property should have a description in the "summary" XML tag or "inheritdoc" XML tag | Warning | Available |
```
Comment (SENya1990) [inline docs/Summary.md:21]: We probably should change the type to `Error` because now PX1007 will report error for mapped DAC field properties of projection DACs with incorrect annotation:
```suggestion
| [PX1007](diagnostics/PX1007.md) | The DAC, DAC extension, or DAC property should have a description in the "summary" XML tag or "inheritdoc" XML tag | Error | Available |
```

File: docs/diagnostics/PX1007.md [hunk 21]
```diff
@@ -5,18 +5,26 @@ This document describes the PX1007 diagnostic.
...
 
+The code fix does one of the following:
  - Adds the `summary` XML tags, in which you type the description of the code item
  - Adds the `exclude` XML tag, which excludes the item from the API and DAC References
+ - Adds the `inheritdoc` tag to a mapped DAC property of a projection DAC.
+   To apply the code fix and add the `inheritdoc` tag to multiple DACs, you can use the **Fix All** button in the code fix menu of Visual Studio. Clicking this button will automatically generate correct documentation for most of the projection DAC properties.
+   This code fix also deletes incorrect documentation tags like `summary` or other incorrectly declared `inheritdoc` tags that should not be placed on the mapped property of a projection DAC.
 
 The new diagnostic is available only if the value of **Tools > Options > Acuminator > Code Analysis > Enable PX1007 diagnostic** is _True_. By default, the value of this option is _False_.
```
Comment (SENya1990) [inline docs/diagnostics/PX1007.md:29]: Maybe we should extend the guide to enable the diagnostic with info from this article but I'm not insisting:
https://wiki.acumatica.com/display/TND/Enabling+PX1007+in+Visual+Studio

File: docs/diagnostics/PX1007.md [hunk 22]
```diff
@@ -28,14 +36,13 @@ The diagnostic is currently displayed for the following public entities:
  - DAC property fields
 
 The diagnostic is not displayed for the following items: 
- - Entities with the `PXHidden`, `PXInternalUseOnly`, or `Obsolete` attribute.
+ - Entities with the `PXInternalUseOnly` or `Obsolete` attribute.
+   ** Note: ** The `PXHidden` attribute does not suppress the diagnostic.
```
Comment (SENya1990) [inline docs/diagnostics/PX1007.md:40]: This is not correct. `PXHidden` attribute suppresses the XP1007 diagnostic but only for DACs. This attribute is applied only to DACs in general. It shouldn't be used on DAC extensions or DAC properties. The changes in this release ensured that the declaration of `PXHidden` on something other than DAC will have no effect.

File: docs/diagnostics/PX1007.md [hunk 23]
```diff
@@ -28,14 +36,13 @@ The diagnostic is currently displayed for the following public entities:
...
  - Members of public entities (except nested public types and DAC property fields).
  - All members of a public entity if the entity has the `exclude` XML tag or is marked with the `PXHidden`, `PXInternalUseOnly`, or `Obsolete` attribute.
```
Comment (SENya1990) [inline docs/diagnostics/PX1007.md:42]: The part with attributes is not precise and became outdated. It should be updated with something like this:
```csharp
 - All members of a public entity if the entity has the `exclude` XML tag or is marked with the `PXInternalUseOnly`, or `Obsolete` attribute. You can also exclude DACs from the documentation by marking them with `PXHidden` attribute.
```

File: docs/diagnostics/PX1007.md [hunk 24]
```diff
@@ -47,9 +54,11 @@ The diagnostic is not displayed for the following items:
    - `LastModifiedByScreenID`
    - `LastModifiedDateTime`
    - `TStamp`
-   - `NoteID`.
+   - `NoteID`
+   - `Attributes`
+   - `GroupMask`
 
-If you suppress this diagnostic with a comment, the default comment message is the following: _Acuminator disable once PX1007 NoXmlCommentForPublicEntityOrDacProperty to be documented later_.
+To remove a DAC or its property from documentation, instead of suppressing the diagnostic, add the `exclude` tag.
```
Comment (SENya1990) [inline docs/diagnostics/PX1007.md:61]: Need to mention DAC extensions too:
```suggestion
To remove a DAC, DAC extension or its property from documentation, instead of suppressing the diagnostic, add the `exclude` tag.
```

**Review by SENya1990 — CHANGES_REQUESTED** [review]
Need to fix the remarks

File: docs/ReleaseNotes.md (hunk 18 above)
Comment (EAndrosova) [reply docs/ReleaseNotes.md:19]: I have mentioned it in the diagnosic description., so added just a small mention here
